src/components/infiniband/linux-infiniband.c

```diff
@@ -335,6 +335,9 @@ add_ib_counter(const char* name, const char* file_name, int extended, ib_device_
     if (new_cnt->ev_name==0 || new_cnt->ev_file_name==0)
     {
         PAPIERROR("cannot allocate memory for counter internal fields");
+	/* free(NULL) is allowed and means no operation is performed */
+	free(new_cnt->ev_name);
```

Comment:
@wcohen Two comments:

- Could you align the two additional free's that you added to be inline with `PAPIERROR`?
- Could you change `free(new_cnt->ev_file_name)` and  `free(new_cnt->ev_name)` to use `papi_free` to follow `papi_free(new_cnt)`? 

Everything else looked good to me. Thank you for running the coverity scan and submitting the changes for a PR.